zebra-rpc/src/methods.rs

```diff
@@ -166,6 +215,33 @@ where
         }
         .boxed()
     }
+
+    fn get_block(&self, height: Height, _verbosity: u8) -> BoxFuture<Result<GetBlock>> {
+        let mut state = self.state.clone();
+
+        async move {
+            let request = zebra_state::Request::Block(zebra_state::HashOrHeight::Height(height));
+            let response = state
+                .ready()
+                .and_then(|service| service.call(request))
+                .await
+                .map_err(|error| Error {
+                    code: ErrorCode::ServerError(0),
+                    message: error.to_string(),
+                    data: None,
+                })?;
+
+            match response {
+                zebra_state::Response::Block(Some(block)) => Ok(GetBlock { data: block.into() }),
+                _ => Err(Error {
+                    code: ErrorCode::ServerError(0),
+                    message: "Block not found".to_string(),
+                    data: None,
+                }),
```

Comment:
Minor: I think we should `panic!` here if the `zebra-state::Response` is not a `Block` variant:

```suggestion
                zebra_state::Response::Block(None) => Err(Error {
                    code: ErrorCode::ServerError(0),
                    message: "Block not found".to_string(),
                    data: None,
                }),
                _ => unreachable!("unmatched response to a block request"),
```

Reply:
I think you are are right, fixed in https://github.com/ZcashFoundation/zebra/pull/3707/commits/6a4055f2030fb896b89f9c015edb6aa0c97be606